tests/features/step_definitions/10209.ACs.feature.steps.intTest.ts

```diff
@@ -0,0 +1,168 @@
+      expect(response.status).toEqual(201);
+    });
+    and('the appropriate audit attributes are set on this new tech record', async () => {
+      responseGET = await request.get(requestUrl + `/${postPayload.vin}/tech-records`);
+      expect(responseGET.body.techRecord[0]).toHaveProperty("createdAt");
+      expect(responseGET.body.techRecord[0]).toHaveProperty("createdByName");
+      expect(responseGET.body.techRecord[0]).toHaveProperty("createdById");
+    });
+    and('the \'statusCode\' of this new tech record is always \'provisional\'', () => {
+      expect(responseGET.body.techRecord[0].statusCode).toEqual("provisional");
+    });
+    and('I am able to POST attributes residing anywhere on the vehicle object', () => {
+      expect(responseGET.body.techRecord[0].grossEecWeight).toEqual(22);
+      expect(responseGET.body.vrms[0].vrm).toEqual("ALKH567");
+      expect(responseGET.body.vrms[0].isPrimary).toEqual(true);
+    });
+    ctx.succeed('done');
+    ctx = null;
+  });
+
+  test('PUT request: HGV vehicle is updated, and the appropriate attributes are automatically set', ({given, when, then, and}) => {
+    let ctx: any = mockContext(opts);
+
+    let requestUrl: string;
+    let response: any;
+    let responseGET: any;
+    let requestUrlGET: string;
+
+    given('I am the vehicles backend service', () => {
+      requestUrl = 'vehicles/ABCDEFGH654321';
+      requestUrlGET = 'vehicles/ABCDEFGH654321/tech-records?status=all';
+    });
+    when('an existing HGV vehicle is updated via the PUT verb', async () => {
+      const putPayload = createPUTPayload();
+      putPayload.techRecord[0].grossEecWeight = 33;
+      response = await request.put(requestUrl).send(putPayload);
+      responseGET = await request.get(requestUrlGET);
+    });
+    then('my PUT action adheres to the HGV validations, present in the linked excel, columns D-E', () => {
+      expect(response.status).toEqual(200);
+      expect(response.body.techRecord.length).toEqual(2);
+    });
+    and('a new identical tech record is created, with the same status, and the updated attributes on it', () => {
+      expect(response.body.techRecord[1].statusCode).toEqual("provisional");
+      expect(response.body.techRecord[1].grossEecWeight).toEqual(33);
+    });
+    and('the previous "pre-update" tech record still exists in DynamoDB, with it\'s status set to archived', () => {
+      expect(response.body.techRecord[0].statusCode).toEqual("archived");
+    });
+    and('the appropriate audit attributes are set on the new updated tech record', () => {
+      expect(responseGET.body.techRecord[1]).toHaveProperty("createdAt");
+      expect(responseGET.body.techRecord[1]).toHaveProperty("createdByName");
+      expect(responseGET.body.techRecord[1]).toHaveProperty("createdById");
+    });
+    and('I am only able to update attributes within the techRecord[] array', () => {
+      expect(response.body.vrms.length).toEqual(2);
```

Comment:
If I send something which attempts to change a field not in the tech record array it will fail with validation error and I cannot test the other "ands". What do you suggest? Maybe do 2 PUT requests and check that it fails when trying to update something outside tech record array

Reply:
In the createPUTPayload function I'm setting different primaryVrm and secondaryVrms, meaning that I'm trying to change those fields, but you're right, I'm not actually checking if they didn't change, will do.

Reply:
New tests look good thanks